src/lib/iframe-tools/identity-manager.js

```diff
@@ -0,0 +1,65 @@
+/* @flow */
+import { setIdentity, getIdentity } from '../local-storage';
+import { getDeviceInfo } from '../get-device-info';
+import { logger } from '../logger';
+
+import { IframeManager } from './iframe-manager';
+
+export class IdentityManager extends IframeManager {
+  constructor(config) {
+    let iframeUrl;
+    
+    if (config.paramsToIdentityUrl) {
+      iframeUrl = config.paramsToIdentityUrl();
+    } else {
+      iframeUrl = 'https://www.paypal.com/muse/identity/index.html';
+    }
+
+    super({ src: iframeUrl });
+
+    this.addMessageListener(this.storeIdentity);
+  }
+
+  onIframeLoad = () => {
+    this.fetchIdentity();
+  }
+
+  logIframeError = (e) => {
+    if (e.data.type !== 'fetch_identity_error') {
+      return;
+    }
+
+    logger.error('identity iframe error:', e.data.payload);
```

Comment:
usually, error objects don't have `data` key, so this may be a bit of a misnomer.
Nothing to do here, just wanted to point that out.

![Screen Shot 2019-11-19 at 3 51 24 PM](https://user-images.githubusercontent.com/686933/69196860-906b6500-0ae4-11ea-8ba8-e1a1bc7dde2a.png)